modules/social_features/social_profile/social_profile.module

```diff
@@ -711,3 +725,17 @@ function social_profile_social_user_account_header_items_alter(array &$menu_item
     ->view($profile, 'small');;
 
 }
+
+/**
+ * Implements hook_block_access().
+ */
+function social_profile_block_access(Block $block, $operation, AccountInterface $account) {
+  // Show statistic block only when new style is enabled.
+  if ($operation == 'view' && $block->getPluginId() === 'profile_statistic_block') {
+    if (theme_get_setting('style') !== 'sky') {
+      return AccessResult::forbidden();
+    }
+  }
+  // No opinion for other situations.
+  return AccessResult::neutral();
+}
```

Comment:
We own the block, so we don't need an external access handler. Additionally, we should decouple this from the theme in some way.